napari/layers/_shapes_layer/shape_util.py

```diff
@@ -3,6 +3,37 @@
 from vispy.geometry import PolygonData
 
 
+def slice_by_plane(data):
+    """Determine if shape is entirely definied within a single 2D plane
+
+    Parameters
+    ----------
+    data : np.ndarray
+        NxD array of vertices specifying the shape.
+
+    Returns
+    ----------
+    key : tuple of int | bool
+        If shape is 2D returns an empty tuple, otherwise return the tuple
+        of int defining the 2D plane if the shape is entirely definied
+        within a single 2D plane. Otherwise return False as the shape is
+        invalid.
+    value : np.ndarray | False
+        Nx2 array of vertices specifying the shape in the 2D plane or False
```

Comment:
put shape in type

Reply:
should it be `value : np.ndarray or False, Nx2` or `value : np.ndarray or False, shape(N,2)` or something else?

Reply:
`value : (N, 2) array or False`

Reply:
perfect! thanks for the help. I need the autoformatter for our docs!

Reply:
actually is `value` even used at all there? it doesn't seem to be

if it's used as a default when it's not passed, however, it shouldn't be defaulted to `False`, but rather `None`, in which case the type would be `value : (N, 2) array, optional`

Reply:
`value` is in the return. It could return as `None` if that was better

Reply:
oh my bad didn't catch that; however `None` is preferable when something exists

although now looking at the other statement, why are you returning two `False`s instead of just raising an exception?

Reply:
I guess I was letting the thing doing the calling decide what to do if no such specification existed (which was to just ignore it) - see in the shapes model file for it being used